trin-core/src/portalnet/overlay.rs

```diff
@@ -175,12 +190,182 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
     }
 
     /// Process overlay uTP listener event
-    pub fn process_utp_event(&self, event: UtpListenerEvent) {
-        // TODO: Handle overlay uTP events
-        debug!(
-            "Got overlay uTP event: {event:?}, protocol id: {:?}",
-            self.protocol
-        );
+    pub fn process_utp_event(&self, event: UtpListenerEvent) -> anyhow::Result<()> {
+        match event {
+            UtpListenerEvent::ClosedStream(utp_payload, _, stream_id) => {
+                // Handle all closed uTP streams, currently we process only AcceptStream here.
+                // FindContent payload is processed explicitly when we send FindContent request.
+                if let AcceptStream(content_keys) = stream_id {
+                    self.process_accept_utp_payload(content_keys, utp_payload)?;
+                }
+            }
+            UtpListenerEvent::ResetStream(..) => {
+                // TODO: Handle reset streams
```

Comment:
Add an issue that's part of a milestone. Alpha?

Reply:
I think `TODO` is confusing here. The intention for this is to be a placeholder if we want to do something with a `Reset` stream in the future. At this point, I think we don't need to do anything in the overlay, the stream is removed from the active connections in `UtpListener`.

I will remove the TODO and just leave it empty for now, no need to create an issue IMO. We may also want to remove this event or refactor the `UtpListener` events but I prefer to do this incrementally.